### PR TITLE
Theming - Add ThemedGlobalStyles

### PR DESCRIPTION
**Pull request by W3stside**
Wraps `GenericLayout` with `ThemedGlobalStyles` and our `ThemeProvider` boy which enables global theming. THere are several changes to make after this, including replacing all `var(...)` with our `theme.colourOfChoice` and others. Will do that on top of this

@alfetopito

### CODE REVIEW

File: src/components/layout/GenericLayout/index.tsx [hunk 1]
```diff
@@ -10,9 +10,12 @@ export interface Props {
...
     <StaticGlobalStyle />
-    <Header menu={menu} tools={navTools} />
-    {children}
-    <Footer />
+    <ThemeProvider>
+      <ThemedGlobalStyle />
+      <Header menu={menu} tools={navTools} />
+      {children}
+      <Footer />
+    </ThemeProvider>
```
Comment (alfetopito) [inline src/components/layout/GenericLayout/index.tsx:18]: Just to be triple sure:

Any children will have the theme and global style already applied, **never again** `<ThemeProvider>` will be used, right?

**Review by alfetopito — APPROVED** [review]

File: src/components/layout/GenericLayout/index.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/layout/GenericLayout/index.tsx:18]: haha yes. you dont need to explicitly pass `ThemeProvider`, you're good.

Well, unless you want to. For example to change the entire theme before passing it in elsewhere. Or to pass a custom theme into it via `theme={customTheme}` prop.

File: src/components/layout/GenericLayout/index.tsx [hunk 2]
```diff
@@ -10,9 +10,12 @@ export interface Props {
 export const GenericLayout: React.FC<Props> = ({ children, menu, navTools }) => (
   <div>
     <StaticGlobalStyle />
-    <Header menu={menu} tools={navTools} />
-    {children}
-    <Footer />
+    <ThemeProvider>
```
Comment (anxolin) [inline src/components/layout/GenericLayout/index.tsx:13]: Them and layout could be decoupled, and each app decides which ThemeProvider to use.
However, I think for now I think is good your current approach, we only have a single ThemeProvider and it's good not to head to ahead of ourselves

**Review by anxolin — APPROVED** [review]

File: src/components/layout/GenericLayout/index.tsx (hunk 2 above)
Comment (W3stside) [reply src/components/layout/GenericLayout/index.tsx:13]: :+1:

**Pull request merged**